lib/matplotlib/spines.py

```diff
@@ -214,16 +214,23 @@ def register_axis(self, axis):
         properties when needed.
         """
         self.axis = axis
-        if self.axis is not None:
-            self.axis.clear()
         self.stale = True
 
     def clear(self):
         """Clear the current spine."""
-        self._position = None  # clear position
+        self._clear()
         if self.axis is not None:
             self.axis.clear()
 
+    def _clear(self):
+        """
+        Clear things directly related to the spine.
+
+        In this way it is possible to avoid clearing the Axis as well when calling
+        from library code where it is known that the Axis is cleared separately.
+        """
+        self._position = None  # clear position
+
```

Comment:
```suggestion
    self.stale = True

```

Reply:
I'll try to do a follow up PR, I suspect most of the `clear` methods do not mark the artist as stale when they should....